src/hotspot/share/gc/shared/pretouchTask.hpp

```diff
@@ -30,15 +30,34 @@
 class PretouchTask : public AbstractGangTask {
   char* volatile _cur_addr;
   char* const _start_addr;
-  char* const _end_addr;
+  char* volatile _end_addr;
   size_t _page_size;
   size_t _chunk_size;
+  volatile uint _n_threads;  // Number of threads participating in pretouch.
+
+  enum TaskStatus{ NotReady, Ready, Done};
+  volatile size_t _task_status;
 
 public:
-  PretouchTask(const char* task_name, char* start_address, char* end_address, size_t page_size, size_t chunk_size);
+  PretouchTask(const char* task_name, char* start_address, char* end_address, size_t page_size, size_t chunk_size, uint n_threads = 0, size_t task_status = Ready);
+
+  void reinitialize(char* start_address, char *end_addr);
+
+  void set_task_status(TaskStatus status) { Atomic::release_store(&_task_status, (size_t)status);  }
+  void set_task_done()                    { Atomic::release_store(&_task_status, (size_t)Done);    }
```

Comment:
There is a convention that when a setter embodies a release_store that release is included in the name e.g. release_set_task_done().

Reply:
OK, will change.